Userland/Applets/ClipboardHistory/ClipboardHistoryModel.cpp

```diff
@@ -111,40 +112,38 @@ void ClipboardHistoryModel::clipboard_content_did_change(ByteString const&)
         add_item(data_and_type);
 }
 
-ErrorOr<void> ClipboardHistoryModel::invalidate_model_and_file(bool rewrite_all)
+ErrorOr<void> ClipboardHistoryModel::invalidate_model_and_file()
 {
     invalidate();
 
-    TRY(write_to_file(rewrite_all));
+    TRY(write_to_file());
     return {};
 }
 
-void ClipboardHistoryModel::add_item(const GUI::Clipboard::DataAndType& item)
+void ClipboardHistoryModel::add_item(GUI::Clipboard::DataAndType const& item)
 {
-    bool has_deleted_an_item = false;
     m_history_items.remove_first_matching([&](ClipboardItem& existing) {
         return existing.data_and_type.data == item.data && existing.data_and_type.mime_type == item.mime_type;
     });
 
-    if (m_history_items.size() == m_history_limit) {
-        m_history_items.take_last();
-        has_deleted_an_item = true;
+    if (m_history_items.size() >= m_history_limit) {
```

Comment:
Not a big deal, but this is not doing the same thing as before.
We were resizing to `m_history_limit - 1` to be able to safely prepend afterwards.
If you're using resize, you should do that after the prepend (and also replace `>=` by `=`)

Reply:
:man_facepalming: Thanks, fixed it!